lib/audit/catalog_to_moab.rb

```diff
@@ -142,4 +143,23 @@ def online_moab_found?(druid, storage_dir)
     false
   end
 
+  # given whether the caller found the expected version of preserved_copy on disk, this will perform
+  # other validations of what's on disk, and will update the status accordingly.
+  def set_status_as_seen_on_disk(found_expected_version)
+    unless moab_validation_errors.empty?
+      update_status(PreservedCopy::INVALID_MOAB_STATUS)
+      return
+    end
+
+    unless found_expected_version
```

Comment:
this is a little less cognitive load ... but for some reason `unless` makes my face scrunch up.  Yet it is awesomely useful.